conf_system/config.txt
################################################################################
##  Raspberry Pi Configuration Settings
##
##  Revision 16, 2013/06/22
##
##  Details taken from the eLinux wiki
##  For up-to-date information please refer to wiki page.
##
##  Wiki Location : http://elinux.org/RPiconfig
##
##
##  Description:
##    Details of each setting are described with each section that begins with
##    a double hashed comment ('##')
##    It is up to the user to remove the single hashed comment ('#') from each
##    option they want to enable, and to set the specific value of that option.
##
##  Overclock settings will be disabled at runtime if the SoC reaches temp_limit
##
################################################################################

################################################################################
##  Standard Definition Video Settings
################################################################################

## sdtv_mode
##     defines the TV standard for composite output
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Normal NTSC (Default)
##     1        Japanese version of NTSC - no pedestal
##     2        Normal PAL
##     3        Brazilian version of PAL - 525/60 rather than 625/50, different
##              subcarrier
##
#sdtv_mode=0

## sdtv_aspect
##     defines the aspect ratio for composite output
##
##     Value    Description
##     -------------------------------------------------------------------------
##     1        4:3 (Default)
##     2        14:9
##     3        16:9
##
#sdtv_aspect=1

## sdtv_disable_colourburst
##     Disables colour burst on composite output. The picture will be
##     monochrome, but possibly sharper
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Colour burst is enabled (Default)
##     1        Colour burst is disabled
##
#sdtv_disable_colourburst=1

################################################################################
##  High Definition Video Settings
################################################################################

## hdmi_safe
##     Use "safe mode" settings to try to boot with maximum hdmi compatibility.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Disabled (Default)
##     1        Enabled (this does: hdmi_force_hotplug=1,
##                                  hdmi_ignore_edid=0xa5000080,
##                                  config_hdmi_boost=4, hdmi_group=2,
##                                  hdmi_mode=4, disable_overscan=0,
##                                  overscan_left=24, overscan_right=24,
##                                  overscan_top=24, overscan_bottom=24)
##
#hdmi_safe=1

## hdmi_force_hotplug
##     Pretends HDMI hotplug signal is asserted so it appears a HDMI display
##     is attached
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Disabled (Default)
##     1        Use HDMI mode even if no HDMI monitor is detected
##
#hdmi_force_hotplug=1

## hdmi_ignore_hotplug
##     Pretends HDMI hotplug signal is not asserted so it appears a HDMI
##     display is not attached
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Disabled (Default)
##     1        Use composite mode even if HDMI monitor is detected
##
#hdmi_ignore_hotplug=1 

## hdmi_drive
##     chooses between HDMI and DVI modes
##
##     Value    Description
##     -------------------------------------------------------------------------
##     1        Normal DVI mode (No sound)
##     2        Normal HDMI mode (Sound will be sent if supported and enabled)
##
#hdmi_drive=2

## hdmi_ignore_edid
##     Enables the ignoring of EDID/display data
##
#hdmi_ignore_edid=0xa5000080

## hdmi_edid_file
##     Read the EDID data from the edid.dat file instead of from the attached
##     device
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Read EDID data from attached device (Default)
##     1        Read EDID data from edid.txt file
##
#hdmi_edid_file=1

## hdmi_ignore_edid_audio
##     Pretends all audio formats are unsupported by display. This means ALSA
##     will default to analogue.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Use EDID provided values (Default)
##     1        Pretend all audio formats are unsupported
##
#hdmi_ignore_edid_audio=1

## hdmi_force_edid_audio
##     Pretends all audio formats are supported by display, allowing
##     passthrough of DTS/AC3 even when not reported as supported.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Use EDID provided values (Default)
##     1        Pretend all audio formats are supported
##
#hdmi_force_edid_audio=1

## hdmi_force_edid_3d
##     Pretends all CEA modes support 3D even when edid doesn't indicate
##     support for them.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Use EDID provided values (Default)
##     1        Pretend 3D mode is supported
##
#hdmi_force_edid_3d=1

## avoid_edid_fuzzy_match
##     Avoid fuzzy matching of modes described in edid.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Use fuzzy matching (Default)
##     1        Avoid fuzzy matching
##
#avoid_edid_fuzzy_match=1

## hdmi_pixel_encoding
##     Force the pixel encoding mode.
##     By default it will use the mode requested from edid so shouldn't
##     need changing.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Use EDID provided values (Default)
##     1        RGB limited   (16-235)
##     2        RGB full      ( 0-255)
##     3        YCbCr limited (16-235)
##     4        YCbCr limited ( 0-255)
##
#hdmi_pixel_encoding=1

## hdmi_group
##     Defines the HDMI type
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Use the preferred group reported by the edid (Default)
##     1        CEA
##     2        DMT
##
#hdmi_group=1

## hdmi_mode
##     defines screen resolution in CEA or DMT format
##
##     H means 16:9 variant (of a normally 4:3 mode).
##     2x means pixel doubled (i.e. higher clock rate, with each pixel repeated
##                                  twice)
##     4x means pixel quadrupled (i.e. higher clock rate, with each pixel
##                                     repeated four times)
##     reduced blanking means fewer bytes are used for blanking within the data
##                      stream (i.e. lower clock rate, with fewer wasted bytes)
##
##     Value    hdmi_group=CEA                  hdmi_group=DMT
##     -------------------------------------------------------------------------
##     1        VGA                             640x350   85Hz
##     2        480p  60Hz                      640x400   85Hz
##     3        480p  60Hz  H                   720x400   85Hz
##     4        720p  60Hz                      640x480   60Hz
##     5        1080i 60Hz                      640x480   72Hz
##     6        480i  60Hz                      640x480   75Hz
##     7        480i  60Hz  H                   640x480   85Hz
##     8        240p  60Hz                      800x600   56Hz
##     9        240p  60Hz  H                   800x600   60Hz
##     10       480i  60Hz  4x                  800x600   72Hz
##     11       480i  60Hz  4x H                800x600   75Hz
##     12       240p  60Hz  4x                  800x600   85Hz
##     13       240p  60Hz  4x H                800x600   120Hz
##     14       480p  60Hz  2x                  848x480   60Hz
##     15       480p  60Hz  2x H                1024x768  43Hz  DO NOT USE
##     16       1080p 60Hz                      1024x768  60Hz
##     17       576p  50Hz                      1024x768  70Hz
##     18       576p  50Hz  H                   1024x768  75Hz
##     19       720p  50Hz                      1024x768  85Hz
##     20       1080i 50Hz                      1024x768  120Hz
##     21       576i  50Hz                      1152x864  75Hz
##     22       576i  50Hz  H                   1280x768        reduced blanking
##     23       288p  50Hz                      1280x768  60Hz
##     24       288p  50Hz  H                   1280x768  75Hz
##     25       576i  50Hz  4x                  1280x768  85Hz
##     26       576i  50Hz  4x H                1280x768  120Hz reduced blanking
##     27       288p  50Hz  4x                  1280x800        reduced blanking
##     28       288p  50Hz  4x H                1280x800  60Hz
##     29       576p  50Hz  2x                  1280x800  75Hz
##     30       576p  50Hz  2x H                1280x800  85Hz
##     31       1080p 50Hz                      1280x800  120Hz reduced blanking
##     32       1080p 24Hz                      1280x960  60Hz
##     33       1080p 25Hz                      1280x960  85Hz
##     34       1080p 30Hz                      1280x960  120Hz reduced blanking
##     35       480p  60Hz  4x                  1280x1024 60Hz
##     36       480p  60Hz  4x H                1280x1024 75Hz
##     37       576p  50Hz  4x                  1280x1024 85Hz
##     38       576p  50Hz  4x H                1280x1024 120Hz reduced blanking
##     39       1080i 50Hz  reduced blanking    1360x768  60Hz
##     40       1080i 100Hz                     1360x768  120Hz reduced blanking
##     41       720p  100Hz                     1400x1050       reduced blanking
##     42       576p  100Hz                     1400x1050 60Hz
##     43       576p  100Hz H                   1400x1050 75Hz
##     44       576i  100Hz                     1400x1050 85Hz
##     45       576i  100Hz H                   1400x1050 120Hz reduced blanking
##     46       1080i 120Hz                     1440x900        reduced blanking
##     47       720p  120Hz                     1440x900  60Hz
##     48       480p  120Hz                     1440x900  75Hz
##     49       480p  120Hz H                   1440x900  85Hz
##     50       480i  120Hz                     1440x900  120Hz reduced blanking
##     51       480i  120Hz H                   1600x1200 60Hz
##     52       576p  200Hz                     1600x1200 65Hz
##     53       576p  200Hz H                   1600x1200 70Hz
##     54       576i  200Hz                     1600x1200 75Hz
##     55       576i  200Hz H                   1600x1200 85Hz
##     56       480p  240Hz                     1600x1200 120Hz reduced blanking
##     57       480p  240Hz H                   1680x1050       reduced blanking
##     58       480i  240Hz                     1680x1050 60Hz
##     59       480i  240Hz H                   1680x1050 75Hz
##     60                                       1680x1050 85Hz
##     61                                       1680x1050 120Hz reduced blanking
##     62                                       1792x1344 60Hz
##     63                                       1792x1344 75Hz
##     64                                       1792x1344 120Hz reduced blanking
##     65                                       1856x1392 60Hz
##     66                                       1856x1392 75Hz
##     67                                       1856x1392 120Hz reduced blanking
##     68                                       1920x1200       reduced blanking
##     69                                       1920x1200 60Hz
##     70                                       1920x1200 75Hz
##     71                                       1920x1200 85Hz
##     72                                       1920x1200 120Hz reduced blanking
##     73                                       1920x1440 60Hz
##     74                                       1920x1440 75Hz
##     75                                       1920x1440 120Hz reduced blanking
##     76                                       2560x1600       reduced blanking
##     77                                       2560x1600 60Hz
##     78                                       2560x1600 75Hz
##     79                                       2560x1600 85Hz
##     80                                       2560x1600 120Hz reduced blanking
##     81                                       1366x768  60Hz
##     82                                       1080p     60Hz
##     83                                       1600x900        reduced blanking
##     84                                       2048x1152       reduced blanking
##     85                                       720p      60Hz
##     86                                       1366x768        reduced blanking
##
#hdmi_mode=1

## config_hdmi_boost
##     configure the signal strength of the HDMI interface.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        (Default)
##     1
##     2
##     3
##     4        Try if you have interference issues with HDMI
##     5
##     6
##     7        Maximum
##
#config_hdmi_boost=0

## hdmi_ignore_cec_init
##     Doesn't sent initial active source message.  Avoids bringing
##     (CEC enabled) TV out of standby and channel switch when rebooting.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Normal behaviour (Default)
##     1        Doesn't sent initial active source message
##
#hdmi_ignore_cec_init=1

## hdmi_ignore_cec
##     Pretends CEC is not supported at all by TV.
##     No CEC functions will be supported.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Normal behaviour (Default)
##     1        Pretend CEC is not supported by TV
##
#hdmi_ignore_cec=1

################################################################################
##  Overscan Video Settings
################################################################################

## overscan_left
##     Number of pixels to skip on left
##
#overscan_left=0

## overscan_right
##     Number of pixels to skip on right
##
#overscan_right=0

## overscan_top
##     Number of pixels to skip on top
##
#overscan_top=0

## overscan_bottom
##     Number of pixels to skip on bottom
##
#overscan_bottom=0

## disable_overscan
##     Set to 1 to disable overscan
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Overscan Enabled (Default)
##     1        Overscan Disabled
##
#disable_overscan=1

################################################################################
##  Framebuffer Video Settings
################################################################################

## framebuffer_width
##     Console framebuffer width in pixels. Default is display width minus
##     overscan.
##
#framebuffer_width=0

## framebuffer_height
##     Console framebuffer height in pixels. Default is display height minus
##     overscan.
##
#framebuffer_height=0

## framebuffer_depth
##     Console framebuffer depth in bits per pixel.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     8        Valid, but default RGB palette makes an unreadable screen
##     16       (Default)
##     24       Looks better but has corruption issues as of 2012/06/15
##     32       Has no corruption issues but needs framebuffer_ignore_alpha=1
##              and shows the wrong colors as of 2012/06/15
##
#framebuffer_depth=16

## framebuffer_ignore_alpha
##     Set to 1 to disable alpha channel. Helps with 32bit.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Enable Alpha Channel (Default)
##     1        Disable Alpha Channel
##
#framebuffer_ignore_alpha=0

################################################################################
##  General Video Settings
################################################################################

## display_rotate
##     Rotate the display clockwise or flip the display.
##     The 90 and 270 degrees rotation options require additional memory on GPU,
##     so won't work with the 16M GPU split.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        0 degrees (Default)
##     1        90 degrees
##     2        180 degrees
##     3        270 degrees
##     0x10000  Horizontal flip
##     0x20000  Vertical flip
##
#display_rotate=0

################################################################################
##  Licensed Codecs
##
##  Hardware decoding of additional codecs can be enabled by purchasing a
##  license that is locked to the CPU serial number of your Raspberry Pi.
##
##  Up to 8 licenses per CODEC can be specified as a comma seperated list.
##
################################################################################

## decode_MPG2
##     License key to allow hardware MPEG-2 decoding.
##
#decode_MPG2=0x12345678

## decode_WVC1
##     License key to allow hardware VC-1 decoding.
##
#decode_WVC1=0x12345678

################################################################################
##  Camera Settings
################################################################################

## disable_camera_led
##     Turn off the red camera led when recording video or taking a still
##     picture.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        LED enabled (Default)
##     1        LED disabled
##
#disable_camera_led=1

################################################################################
##  Test Settings
################################################################################

## test_mode
##     Enable test sound/image during boot for manufacturing test.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Disable Test Mod (Default)
##     1        Enable Test Mode
##
#test_mode=0

################################################################################
##  Memory Settings
################################################################################

## disable_l2cache
##     Disable arm access to GPU's L2 cache. Needs corresponding L2 disabled
##     kernel.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Enable L2 Cache (Default)
##     1        Disable L2 cache
##
#disable_l2cache=0

## gpu_mem
##     GPU memory allocation in MB for all board revisions.
##
##     Default 64
##
#gpu_mem=128

## gpu_mem_256
##     GPU memory allocation in MB for 256MB board revision.
##     This option overrides gpu_mem.
##
#gpu_mem_256=128

## gpu_mem_512
##     GPU memory allocation in MB for 512MB board revision.
##     This option overrides gpu_mem.
##
#gpu_mem_512=128

## disable_pvt
##     Disable adjusting the refresh rate of RAM every 500ms
##     (measuring RAM temparature).
##
#disable_pvt=1

################################################################################
##  CMA - Dynamic Memory Split
##
##  CMA enables dynamic management of the ARM and GPU memory split at runtime.
##
##  The following options need to be in cmdline.txt for CMA to work:
##    coherent_pool=6M smsc95xx.turbo_mode=N
##
################################################################################

## cma_lwm
##     When GPU has less than cma_lwm (low water mark) memory available it
##     will request some from ARM.
##
#cma_lwm=16

## cma_hwm
##     When GPU has more than cma_hwm (high water mark) memory available it
##     will release some to ARM.
##
#cma_hwm=32

################################################################################
##  Boot Option Settings
################################################################################

## disable_commandline_tags
##     Stop start.elf from filling in ATAGS (memory from 0x100) before
##     launching kernel
##
#disable_commandline_tags=0

## cmdline (string)
##     Command line parameters. Can be used instead of cmdline.txt file
##
#cmdline=""

## kernel (string)
##     Alternative name to use when loading kernel.
##
#kernel=""

## kernel_address
##     Address to load kernel.img file at
##
#kernel_address=0x00000000

## kernel_old
##     Support loading old kernels
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Disabled (Default)
##     1        Load kernel at address 0x00000000
##
#kernel_old=1

## ramfsfile (string)
##     ramfs file to load
##
#ramfsfile=""

## ramfsaddr
##     Address to load ramfs file at
##
#ramfsaddr=0x00000000

## initramfs (string address)
##     ramfs file and address to load it at (it's like ramfsfile+ramfsaddr in
##     one option).
##
##     NOTE: this option uses different syntax than all other options - you
##           should not use "=" character here.
##
#initramfs initramf.gz 0x00800000

## device_tree_address
##     Address to load device_tree at
##
#device_tree_address=0x00000000

## init_uart_baud
##     Initial uart baud rate.
##
##     Default 115200
##
#init_uart_baud=115200

## init_uart_clock
##     Initial uart clock.
##
##     Default 3000000 (3MHz)
##
#init_uart_clock=3000000

## init_emmc_clock
##     Initial emmc clock, increasing this can speedup your SD-card.
##
##     Default 100000000 (100mhz)
##
#init_emmc_clock=100000000

## boot_delay
##     Wait for a given number of seconds in start.elf before loading
##     kernel.img.
##
##     delay = (1000 * boot_delay) + boot_delay_ms
##
##     Default 1
##
#boot_delay=0

## boot_delay_ms
##     Wait for a given number of milliseconds in start.elf before loading
##     kernel.img.
##
##     delay = (1000 * boot_delay) + boot_delay_ms
##
##     Default 0
##
#boot_delay_ms=0

## avoid_safe_mode
##     Adding a jumper between pins 5 & 6 of P1 enables a recovery Safe Mode.
##     If pins 5 & 6 are used for connecting to external devices (e.g. GPIO),
##     then this setting can be used to ensure Safe Mode is not triggered.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Respect Safe Mode input (Default)
##     1        Ignore Safe Mode input
##
#avoid_safe_mode=1

## disable_splash
##     Avoids the rainbow splash screen on boot.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Splash screen enabled (Default)
##     1        Splash screen disabled
##
#disable_splash=1

################################################################################
##  Overclocking Settings
##
##  ARM, SDRAM and GPU each have their own PLLs and can have unrelated
##  frequencies.
##
##  The GPU core, h264, v3d and isp share a PLL, so need to have related
##  frequencies.
##  pll_freq = floor(2400 / (2 * core_freq)) * (2 * core_freq)
##  gpu_freq = pll_freq / [even number]
##
##  The effective gpu_freq is automatically rounded to nearest even integer, so
##  asking for core_freq = 500 and gpu_freq = 300 will result in divisor of
##  2000/300 = 6.666 => 6 and so 333.33MHz.
##
##
##  Standard Profiles:
##                  arm_freq    core_freq    sdram_freq    over_voltage
##     -------------------------------------------------------------------------
##     None         700         250          400           0
##     Modest       800         300          400           0
##     Medium       900         333          450           2
##     High         950         450          450           6
##     Turbo        1000        500          500           6
##
################################################################################

## force_turbo
##     Control the kernel "ondemand" governor. It has no effect if no overclock
##     settings are specified.
##     May set warrany bit.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Enable dynamic clocks and voltage for the ARM core, GPU core and
##              SDRAM (Default).
##              Overclocking of h264_freq, v3d_freq and isp_freq is ignored.
##     1        Disable dynamic clocks and voltage for the ARM core, GPU core
##              and SDRAM.
##              Overclocking of h264_freq, v3d_freq and isp_freq is allowed.
##
#force_turbo=0

## initial_turbo
##     Enables turbo mode from boot for the given value in seconds (up to 60)
##     or until cpufreq sets a frequency. Can help with sdcard corruption if
##     overclocked.
##
##     Default 0
##
#initial_turbo=0

## temp_limit
##     Overheat protection. Sets clocks and voltages to default when the SoC
##     reaches this Celsius value.
##     Setting this higher than default voids warranty.
##
##     Default 85
##
#temp_limit=85

## arm_freq
##     Frequency of ARM in MHz.
##
##     Default 700.
##
#arm_freq=700

## arm_freq_min
##     Minimum frequency of ARM in MHz (used for dynamic clocking).
##
##     Default 700.
##
#arm_freq_min=700

## gpu_freq
##     Sets core_freq, h264_freq, isp_freq, v3d_freq together.
##
##     Default 250.
##
#gpu_freq=250

## core_freq
##     Frequency of GPU processor core in MHz. It has an impact on ARM
##     performance since it drives L2 cache.
##
##     Default 250.
##
#core_freq=250

## core_freq_min
##     Minimum frequency of GPU processor core in MHz (used for dynamic
##     clocking). It has an impact on ARM performance since it drives L2 cache.
##
##     Default 250.
##
#core_freq_min=250

## h264_freq
##     Frequency of hardware video block in MHz.
##
##     Default 250.
##
#h264_freq=250

## isp_freq
##     Frequency of image sensor pipeline block in MHz.
##
##     Default 250.
##
#isp_freq=250

## v3d_freq
##     Frequency of 3D block in MHz.
##
##     Default 250.
##
#v3d_freq=250

## sdram_freq
##     Frequency of SDRAM in MHz.
##
##     Default 400.
##
#sdram_freq=400

## sdram_freq_min
##     Minimum frequency of SDRAM in MHz (used for dynamic clocking).
##
##     Default 400.
##
#sdram_freq_min=400

## avoid_pwm_pll
##     Don't dedicate a pll to PWM audio. This will reduce analogue audio
##     quality slightly. The spare PLL allows the core_freq to be set
##     independently from the rest of the gpu allowing more control over
##     overclocking.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     0        Linked core_freq (Default)
##     1        Unlinked core_freq
##
#avoid_pwm_pll=1

################################################################################
##  Voltage Settings
################################################################################

## current_limit_override
##     Disables SMPS current limit protection. Can help if you are currently
##     hitting a reboot failure when overclocking too high.
##     May set warrany bit.
##
#current_limit_override=0x5A000020

## over_voltage
##     ARM/GPU core voltage adjust.
##     May set warrany bit.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     -16      0.8   V
##     -15      0.825 V
##     -14      0.85  V
##     -13      0.875 V
##     -12      0.9   V
##     -11      0.925 V
##     -10      0.95  V
##     -9       0.975 V
##     -8       1.0   V
##     -7       1.025 V
##     -6       1.05  V
##     -5       1.075 V
##     -4       1.1   V
##     -3       1.125 V
##     -2       1.15  V
##     -1       1.175 V
##     0        1.2   V (Default)
##     1        1.225 V
##     2        1.25  V
##     3        1.275 V
##     4        1.3   V
##     5        1.325 V
##     6        1.35  V
##     7        1.375 V (requires force_turbo=1 or current_limit_override)
##     8        1.4   V (requires force_turbo=1 or current_limit_override)
##
#over_voltage=0

## over_voltage_min
##     Minimum ARM/GPU core voltage adjust (used for dynamic clocking).
##
##     Value    Description
##     -------------------------------------------------------------------------
##     -16      0.8   V
##     -15      0.825 V
##     -14      0.85  V
##     -13      0.875 V
##     -12      0.9   V
##     -11      0.925 V
##     -10      0.95  V
##     -9       0.975 V
##     -8       1.0   V
##     -7       1.025 V
##     -6       1.05  V
##     -5       1.075 V
##     -4       1.1   V
##     -3       1.125 V
##     -2       1.15  V
##     -1       1.175 V
##     0        1.2   V (Default)
##     1        1.225 V
##     2        1.25  V
##     3        1.275 V
##     4        1.3   V
##     5        1.325 V
##     6        1.35  V
##     7        1.375 V (requires force_turbo=1)
##     8        1.4   V (requires force_turbo=1)
##
#over_voltage_min=0

## over_voltage_sdram
##     Sets over_voltage_sdram_c, over_voltage_sdram_i, over_voltage_sdram_p
##     together
##
##     Value    Description
##     -------------------------------------------------------------------------
##     -16      0.8   V
##     -15      0.825 V
##     -14      0.85  V
##     -13      0.875 V
##     -12      0.9   V
##     -11      0.925 V
##     -10      0.95  V
##     -9       0.975 V
##     -8       1.0   V
##     -7       1.025 V
##     -6       1.05  V
##     -5       1.075 V
##     -4       1.1   V
##     -3       1.125 V
##     -2       1.15  V
##     -1       1.175 V
##     0        1.2   V (Default)
##     1        1.225 V
##     2        1.25  V
##     3        1.275 V
##     4        1.3   V
##     5        1.325 V
##     6        1.35  V
##     7        1.375 V
##     8        1.4   V
##
#over_voltage_sdram=0

## over_voltage_sdram_c
##     SDRAM controller voltage adjust.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     -16      0.8   V
##     -15      0.825 V
##     -14      0.85  V
##     -13      0.875 V
##     -12      0.9   V
##     -11      0.925 V
##     -10      0.95  V
##     -9       0.975 V
##     -8       1.0   V
##     -7       1.025 V
##     -6       1.05  V
##     -5       1.075 V
##     -4       1.1   V
##     -3       1.125 V
##     -2       1.15  V
##     -1       1.175 V
##     0        1.2   V (Default)
##     1        1.225 V
##     2        1.25  V
##     3        1.275 V
##     4        1.3   V
##     5        1.325 V
##     6        1.35  V
##     7        1.375 V
##     8        1.4   V
##
#over_voltage_sdram_c=0

## over_voltage_sdram_i
##     SDRAM I/O voltage adjust.
##
##     Value    Description
##     -------------------------------------------------------------------------
##     -16      0.8   V
##     -15      0.825 V

gpio_poller/config.h

//#define DEBUG          //uncomment to enable debugging output

//-----------------defines------------------------------
// Timings
#define WAIT_CYCLE  1   //in miliseconds (must be a common divisor of the following 3 defines)
#define ENC_READ_TIME 1 //in miliseconds
#define ENC_SEND_TIME 20 //in miliseconds
#define KEY_READ_TIME 50 //in miliseconds
 
#define LONG_PRESS_TIME 500 //in miliseconds

// Display
#define DISPLAY  ":0"   //NULL = user environment variable DISPLAY. Otherwise pass a string for the display host:displaynumber e.g. :0

// Rotary Encoder
#define ROTARY_ENC_PIN_A    RPI_V2_GPIO_P1_18 // Pin A
#define ROTARY_ENC_PIN_B    RPI_V2_GPIO_P1_16 // Pin B

// Keys, Rotary Encoder Keyboard events
#define ROTARY_ENC_KEYCODE_LEFT XK_Up // Scroll Up (Rotary Encoder)
#define ROTARY_ENC_KEYCODE_RIGHT XK_Down // Scroll Down (Rotary Encoder)


//----------------calculations------------------------

#define KEY_READ_CYCLES ((KEY_READ_TIME)/(WAIT_CYCLE)) //how many cycles to wait to read the keys
#define LONG_PRESS_CYCLES ((LONG_PRESS_TIME)/(WAIT_CYCLE)/(KEY_READ_CYCLES)) //how many times do we need to read the keys to define a longpress
#define ENC_READ_CYCLES ((ENC_READ_TIME)/(WAIT_CYCLE)) //how many cycles to wait to read the encoder
#define ENC_SEND_CYCLES ((ENC_SEND_TIME)/(WAIT_CYCLE)) //how many cycles to wait to send the encoder state
#define NUM_PINS (sizeof(pin_configuration)/ sizeof(pin_setting_t)) //how many pins are configured

//----------------structures --------------------------
#include <stdbool.h> // for bool datatype
#include <stdint.h>  //for uint_*t and int_*t datatypes
#include <bcm2835.h> // Library for of the BCM2835 SOC
#include <X11/keysym.h> // Headerfile for the X11 keycode defines

typedef struct pin_setting_s {
	uint8_t pin;   //the pin to press. one of RPI_V2_GPIO_P1_XX
	unsigned int keycode_short; //the keycode to send for a short press event
	unsigned int keycode_long; //the keycode to send for a long press event

} pin_setting_t;




//-------- Assign Keyboard events to pins ---------------------
//see \usr\include\X11\keysymdef.h for the keycode defines
const pin_setting_t pin_configuration[] = {

							//PIN# Command Short, Command Long
							//01 Pin #1 = +3.3V (Primary
							//02 Pin #2 = +5V (Primary)
// { RPI_V2_GPIO_P1_03, 0 , 0 }, 			//03 Not in use, Not in use
							//04 Pin #4 = +5V
// { RPI_V2_GPIO_P1_05, 0 , 0 }, 			//05 Not in use, Not in use
							//06 Pin #6 = Ground
 { RPI_V2_GPIO_P1_07, XK_minus , XK_minus }, 		//07 Vol -, Same as short
// { RPI_V2_GPIO_P1_08, 0 , 0 }, 			//08 UART
							//09 Pin #9 = Ground
// { RPI_V2_GPIO_P1_10, 0 , 0 }, 			//10 UART
 { RPI_V2_GPIO_P1_11, XK_plus, XK_plus }, 		//11 Vol +, Same as short
// { RPI_V2_GPIO_P1_12, 0 , 0 }, 			//12 Not in use, Not in use
 { RPI_V2_GPIO_P1_13, XK_c, 0 }, 			//13 Pause, -
							//14 Pin #14 = Ground
 { RPI_V2_GPIO_P1_15, XK_Return, 0 },			//15 Enter (Go), Not in use
// { RPI_V2_GPIO_P1_16, 0 , 0 }, 			//16 Scroll Down (Rotary Encoder)
							//17 Pin #17 = +3.3V
// { RPI_V2_GPIO_P1_18, 0 , 0 }, 			//18 Scroll Up (Rotary Encoder)
 { RPI_V2_GPIO_P1_19, XK_a, XK_A }, 			//19 Add, Add (hold)
							//20 Pin #20 = Ground
 { RPI_V2_GPIO_P1_21, XK_p, XK_P }, 			//21 Play, -
 { RPI_V2_GPIO_P1_22, XK_b, XK_B }, 			//22 Skip Track (Forward), Fwd (hold)
 { RPI_V2_GPIO_P1_23, XK_Escape, XK_h }, 		//23 Back, Home
 { RPI_V2_GPIO_P1_24, XK_z, XK_Z }, 			//24 Skip Track (Back), Rew (hold)
							//25 Pin #25 = Ground
 { RPI_V2_GPIO_P1_26, XK_bracketleft, XK_bracketright } //26 Now Playing, Show Playlist
};



